src/main/java/org/apache/maven/shared/dependency/analyzer/asm/ConstantPoolParser.java

```diff
@@ -142,9 +141,8 @@
                     buf.getChar();
                     buf.getChar();
                     break;
+                case CONSTANT_STRING:
```

Comment:
This switch pattern is error prone. It's more reliable against future changes if every case has its own code even when they're the same. If necessary, extract a helper function.

Reply:
Thank you! I'll fix that.